rust/yarp/build.rs

```diff
@@ -243,7 +243,8 @@ fn write_node(file: &mut File, node: &Node) -> Result<(), Box<dyn std::error::Er
             NodeParamType::OptionalLocation => {
                 writeln!(file, "    pub fn {}(&self) -> Option<Location<'pr>> {{", node_param.name)?;
                 writeln!(file, "        let pointer: *mut yp_location_t = unsafe {{ &mut (*self.pointer).{} }};", node_param.name)?;
-                writeln!(file, "        if pointer.is_null() {{")?;
+                writeln!(file, "        let start: *const u8 = unsafe {{ (*pointer).start }};")?;
```

Comment:
Can you get rid of the extra type here? In case it changes in the future, I don't think we need to be quite this explicit

Reply:
Sure, done!